tests/test_course.py

```diff
@@ -1,15 +1,120 @@
 import pytest
+from tests.base_tester import BaseTester
 
 
-class TestAdminCourse:
+class TestAdminCourse(BaseTester):
     '''Test courses panel used my admins
     '''
-    def test_without_username_and_password(self, client):
-        # Signup without username and password
-        rv = client.post('/auth/signup', json={'password': 'test'})
+    def test_view(self, client_admin):
+        # Get all courses
+        rv = client_admin.get('/course')
         json = rv.get_json()
+        assert json['message'] == 'Success.'
+        assert rv.status_code == 200
+        assert json['status'] == 'ok'
+        assert json['data'] == []
+
+    def test_add_with_invalid_username(self, client_admin):
+        # add a course with non-existent username
+        rv = client_admin.post('/course',
+                               json={
+                                   'course': 'math',
+                                   'teacher': 'adminn'
+                               })
+        json = rv.get_json()
+        assert json['message'] == 'User not found.'
+        assert rv.status_code == 404
+        assert json['status'] == 'err'
+
+    def test_add(self, client_admin):
```

Comment:
why you don't just call it `test_add_course` or something told me what this function is test for?

Reply:
Because it is in the class name?